data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.plugins.source.configuration.SqsOptions;
+import com.amazonaws.services.s3.event.S3EventNotification;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.Message;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SqsException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SqsWorker implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(SqsWorker.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private S3SourceConfig s3SourceConfig;
+    private SqsClient sqsClient;
+    private S3Service s3Service;
+    private SqsOptions sqsOptions;
+
+    public SqsWorker(final SqsClient sqsClient, final S3Service s3Service, final S3SourceConfig s3SourceConfig) {
+        this.s3SourceConfig = s3SourceConfig;
+        this.sqsClient = sqsClient;
+        this.s3Service = s3Service;
+        sqsOptions = s3SourceConfig.getSqsOptions();
+    }
+
+    @Override
+    public void run() {
+
+        while(true) {
+            List<Message> messages = new ArrayList<>();
+
+            try {
+                ReceiveMessageRequest receiveMessageRequest = createReceiveMessageRequest();
+                messages.addAll(sqsClient.receiveMessage(receiveMessageRequest).messages());
+            } catch (SqsException e) {
+                LOG.error("Error reading from SQS: {}", e.awsErrorDetails().errorMessage());
+            }
+
+            // read each message as S3 event message
+            Map<Message, S3EventNotification.S3EventNotificationRecord> s3EventNotificationRecords = messages.stream()
+                    .collect(Collectors.toMap(message -> message, this::convertS3EventMessages));
+
+            // build s3ObjectPointer from S3EventNotificationRecord if event name starts with ObjectCreated
+            List<S3ObjectReference> addedObjects = new ArrayList<>();
+            for (Map.Entry<Message, S3EventNotification.S3EventNotificationRecord> entry: s3EventNotificationRecords.entrySet()) {
+                if (entry.getValue().getEventName().startsWith("ObjectCreated")) {
```

Comment:
I'd like to create an `S3EventFilter` structure. There are couple motivations I have in this:

1. We are also going to add an optional user-configuration to support filtering by bucketName or accountId.
2. If we add SNS as an option in future iterations, this filter should be the same code.


Here is a possible design:

```
public interface S3EventFilter {
  Optional<S3EventNotification.S3EventNotificationRecord> filter(S3EventNotification.S3EventNotificationRecord notification)
}

class ObjectCreatedFilter implements S3EventFilter {
  Optional<S3EventNotification.S3EventNotificationRecord> filter(S3EventNotification.S3EventNotificationRecord entry) {
    if (entry.getValue().getEventName().startsWith("ObjectCreated"))
      return Optional.of(entry);
    else
      return Optional.empty();
  }
}
```

We can build more complicated filters for the bucketName and accountId in another PR.